pkg/shp/cmd/build/upload.go

```diff
@@ -106,8 +119,28 @@ func (u *UploadCommand) Complete(p *params.Params, _ *genericclioptions.IOStream
 	if err != nil {
 		return err
 	}
+	shpClientSet, err := p.ShipwrightClientSet()
+	if err != nil {
+		return err
+	}
+
+	// check that the cluster actually contains a build with this name
+	build, err := shpClientSet.ShipwrightV1alpha1().Builds(p.Namespace()).Get(u.cmd.Context(), u.buildRefName, metav1.GetOptions{})
```

Comment:
Could we reuse `bundle.GetSourceBundleImage()` here? This method could return the bundle image (string) and error, so when the bundle image is empty the upload method is "streaming" otherwise "bundling".

Reply:
Will look into it.

Reply:
The `GetSourceBundleImage` function takes a `BuildRun` as input. At this point, we have a build name.